***В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча.***
---
- Какова вероятность того, что все мячи белые?

В этой задаче вероятность нам неизвестна, поэтому никакие распределения тут не участвуют. 

Это стандартная задача на сочетания

Нам известно, что в первой корзине из 10 мячей - 7 белые и то, что мы берем 2 мяча из каждой.

Соответственно сочетания:
- C(7,2)
- C(3,0)

Во второй корзине аналогично:
- C(9,2)
- C(2,0)

In [2]:
from math import factorial

white_first = factorial(7) / (factorial(2) * factorial(7 - 2))
white_second = factorial(9) / (factorial(2) * factorial(9 - 2))
without_white_first = without_white_second = 1

white_first, white_second

(21.0, 36.0)

2) Определяем общее количество комбинаций

In [3]:
full_first = factorial(10) / (factorial(2) * factorial(10 - 2))
full_second = factorial(11) / (factorial(2) * factorial(11 - 2))

full_first, full_second

(45.0, 55.0)

3) Определяем вероятности для мячей из отдельных корзин

In [4]:
p_1 = (white_first * without_white_first) / full_first
p_2 = (white_second * without_white_second) / full_second

p_1, p_2

(0.4666666666666667, 0.6545454545454545)

4) Найдем общую вероятность получения всех четырех белых шаров

In [5]:
p_1 * p_2

0.3054545454545455

### **Ответ: Вероятность, что все 4 шара окажутся белыми = 30.5%**

- Какова вероятность того, что ровно два мяча белые? 

1) Найдем сочетания для:
- 1 и 1
- 2 и 0
- 0 и 2

In [13]:
def combination(n, k):
    return factorial(n) / (factorial(k) * factorial(n - k))


def probability(c, full):
    return c / full


one_f = combination(7, 1) * combination(3, 1)
one_s = combination(9, 1) * combination(2, 1)

two_f = combination(7, 2) 
two_s = combination(2, 2) 

three_f = combination(3, 2)
three_s = combination(9, 2)


first = probability(one_f, full_first) * probability(one_s, full_second)
second = probability(two_f, full_first) * probability(two_s, full_second)
third = probability(three_f, full_first) * probability(three_s, full_second)

x = combination(5, 2)


first + second + third

(0.20484848484848486, 10.0)

### **Ответ: Вероятность того, что ровно 2 мяча белые = 20.5%**

- Какова вероятность того, что хотя бы один мяч белый?

В этой задаче необходимо воспользоваться обратной вероятностью, а не считать все возможные вариации, как в прошлой задаче.

In [7]:
one = combination(3, 2) # 2 других из 3 других первой корзины
two = combination(2, 2) # 2 других из 2 других второй корзины

x = probability(one, full_first)
y = probability(two, full_second)

1 - (x * y)

0.9987878787878788

### **Ответ: Вероятность того, что хотя бы 1 мяч окажется белым = 99.88%**